lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -221,9 +221,17 @@ bool BorrowingOperand::visitLocalEndScopeUses(
   case BorrowingOperandKind::TryApply:
   case BorrowingOperandKind::Yield:
     return true;
-  case BorrowingOperandKind::Branch:
+  case BorrowingOperandKind::Branch: {
+    auto *br = cast<BranchInst>(op->getUser());
+    for (auto *use : br->getArgForOperand(op)->getUses())
+      if (use->isLifetimeEnding())
+        if (!func(use))
+          return false;
     return true;
   }
+  }
+
+  llvm_unreachable("Covered switch isn't covered");
```

Comment:
The reason to have this is so that if someone is lazy, ignores the compiler, we get an assert. I don't think it hurts and also shows that the code writer intended this code to always be covered. So I think it is reasonable.